platform/admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBeanTest.java

```diff
@@ -597,4 +704,51 @@ public void testGetPluginsForApplication() throws Exception {
         serviceBean.getPluginsForApplication(TEST_APP_NAME).get(0),
         is(plugin1JSON));
   }
+
+  @Test
+  public void testRestartWithKarafRestart() throws Exception {
+    final ApplicationServiceBean serviceBean =
+        new ApplicationServiceBean(
+            testAppService,
+            testConfigAdminExt,
+            mBeanServer,
+            mockFeaturesService,
+            mockSystemService);
+
+    serviceBean.restart();
+
+    assertThat(
+        "Restart system property was cleared", System.getProperty(RESTART_JVM), equalTo("false"));
```

Comment:
I'm a bit confused here why we're asserting that the `karaf.restart.jvm` is set to false when the test name says it's restarting with the Karaf Restart. Maybe I'm misinterpreting the purpose of this test. It looks like it's testing that the Karaf restart service is not being reused but the name sounds the opposite

Reply:
it is using karaf to halt and restart vs using the wrapper

Reply:
Then I thought `karaf.restart.jvm` property should be true?

Reply:
if it was true then karaf would be restarting. When using karaf we are trying to exit out of karaf which is why the property is set to false. 
- So when using the wrapper, we don't care as we asl the wrapper to do it.
- When using karaf scripts, we bypass the normal karaf restart which would be triggered with a `true` such that it will exit out of the karaf script back to the ddf script where we look for the restart.jvm file and restart everything

Reply:
Ah okay I get it now. Because we're just telling Karaf to halt

Reply:
Yup. When karaf halts (either because you ask it to reboot or because you call `halt()`, it checks that system property and if it is set to `true` it exits with a special error code of 10 which the karaf script translates into restarting the whole JVM. If you don't set it or if it is `false`, a reboot will be caught internally in Java by karaf and it will restart the container using the same JVM. If you call `halt()`, it will exit normally and the karaf script will eit also normally and then our ddf script will check the restart.jvm file and decide to restart everything instead of exiting normally. We were using that special file already when we detected an out of memory condition to kill the corrupted ddf process and have our ddf script automatically restart.